RA-13

KinCadeFoster opened the pull request:
- Добавил тэг NestedScrollView в качестве основного контейнера
- Реализовал в шапке изображение и название рецепта, стилизовал согласно макету
- Добавил заголовки для секций 'Ингредиенты' и 'Способ приготовления'
- Добавил два контейнера RecyclerView для соответствующих списков (rvIngredients и rvMethod)
- Создал адаптер для ингредиентов IngredientsAdapter и шагов приготовления MethodAdapter
- Реализовал горизонтальный разделитель между элементами списков
- Оптимизировал код

KinCadeFoster committed:
RA-13
- Добавил тэг NestedScrollView в качестве основного контейнера
- Реализовал в шапке изображение и название рецепта, стилизовал согласно макету
- Добавил заголовки для секций 'Ингредиенты' и 'Способ приготовления'
- Добавил два контейнера RecyclerView для соответствующих списков (rvIngredients и rvMethod)
- Создал адаптер для ингредиентов IngredientsAdapter и шагов приготовления MethodAdapter
- Реализовал горизонтальный разделитель между элементами списков
- Оптимизировал код

ievetrovdev commented on app/src/main/java/ru/example/recipeapp/RecipeFragment.kt:
```diff
@@ -43,6 +51,40 @@ class RecipeFragment : Fragment() {
+            LinearLayoutManager.VERTICAL
+        ).apply {
+            dividerThickness = resources.getDimensionPixelSize(R.dimen.divider_thickness)
+            dividerColor = resources.getColor(R.color.divider_color, null)
```
Щас есть несколько проблем с divider:
Он занимает полную длину вьюхи, на дизайне такого нет
Для последнего элемента полоса не нужна

Нужно настроить эти параметры
isLastItemDecorated
setDividerInsetStartResource
setDividerInsetEndResource

ievetrovdev commented on app/src/main/res/layout/fragment_recipe.xml:
```diff
@@ -1,16 +1,83 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:nestedScrollingEnabled="false"
+            android:layout_margin="@dimen/margin_16"
```
Не хватает background ввиде shape другого цвета

ievetrovdev commented on app/src/main/res/layout/fragment_recipe.xml:
```diff
@@ -1,16 +1,83 @@
+            android:id="@+id/rvMethod"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:nestedScrollingEnabled="false"
```
Не хватает background ввиде shape другого цвета

ievetrovdev commented on app/src/main/res/layout/item_ingredient.xml:
```diff
@@ -0,0 +1,13 @@
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical"
+    android:padding="8dp">
```
Вынести в ресурсы

ievetrovdev commented on app/src/main/res/layout/item_ingredient.xml:
```diff
@@ -0,0 +1,13 @@
+    android:padding="8dp">
+
+    <TextView
+        android:id="@+id/tvIngredient"
```
Шрифт не соответствует дизайну

ievetrovdev commented on app/src/main/res/layout/item_method.xml:
```diff
@@ -0,0 +1,13 @@
+    android:padding="8dp">
+
+    <TextView
+        android:id="@+id/tvStep"
```
Шрифт не соответствует дизайну

ievetrovdev commented on app/src/main/res/layout/item_method.xml:
```diff
@@ -0,0 +1,13 @@
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical"
+    android:padding="8dp">
```
Вынести в ресурсы

ievetrovdev commented on app/src/main/res/layout/item_ingredient.xml:
```diff
@@ -0,0 +1,13 @@
+        android:id="@+id/tvIngredient"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/ingredients" />
```
Не хватает еще TextView для показа величины и количества

ievetrovdev commented on gradle/libs.versions.toml:
```diff
@@ -21,6 +22,7 @@ androidx-appcompat = { group = "androidx.appcompat", name = "appcompat", version
 material = { group = "com.google.android.material", name = "material", version.ref = "material" }
 androidx-activity = { group = "androidx.activity", name = "activity", version.ref = "activity" }
 androidx-constraintlayout = { group = "androidx.constraintlayout", name = "constraintlayout", version.ref = "constraintlayout" }
+androidx-legacy-support-v4 = { group = "androidx.legacy", name = "legacy-support-v4", version.ref = "legacySupportV4" }
```
Почему решил добавить эту библиотеку? В ней содержится устаревший legacy код, надо стараться избегать его и не тянуть в новые проекты

KinCadeFoster replied:
Видимо эксперементировал, осталось, сам не помню в какой момент добавлял

KinCadeFoster committed:
RA-13
- правки по замечаниям

ievetrovdev commented on app/src/main/res/values/dimens.xml:
```diff
@@ -7,14 +7,20 @@
     <dimen name="text_size_20">20sp</dimen>
+
+    <dimen name="divider_thickness">1dp</dimen>
+    <dimen name="divider_inset_start">12dp</dimen>
```
Плохим тоном является дублирование ресурсов, это касается и цветов, и строк, и dimens.  У тебя повторяетсся 12dp, 4dp. Также ссылаться на item и плодить много одинаковых значений тоже не очень правильно. Создай просто common_8dp как dimen и используй во всех местах напрямую.

KinCadeFoster committed:
RA-13
- понизил версию min sdk (что бы можно было тестировать на телефоне)
- рефакторинг dimens
